lib/src/date_picker_theme.dart

```diff
@@ -20,7 +20,7 @@ const double DATETIME_PICKER_ITEM_HEIGHT = 36.0;
 const TextStyle DATETIME_PICKER_ITEM_TEXT_STYLE =
     const TextStyle(color: Color(0xFF000046), fontSize: 16.0);
 
-class DateTimePickerTheme extends Diagnosticable {
+class DateTimePickerTheme with Diagnosticable {
```

Comment:
To support both stable and beta channels, I'd suggest that this should be 
```dart
class DateTimePickerTheme with DiagnosticableMixin { ... }
``` 
until `DiagnosticableMixin` is officially deprecated.

https://github.com/flutter/flutter/issues/50498